### ISSUE TITLE
Custom upstream DNS field specification

### ISSUE THREAD

**Issue by jamilbk** [CloudFire-LLC/cloudfire#2297]
DNS is complicated, especially taking DoT and DoH configuration into account. Let's make some assumptions to make validations easier among the 3 main types of DNS servers.

We could use embedded schemas for the 3 types of DNS servers we support: `ip_address`, `dns_over_tls`, `dns_over_https`

#### `ip_address`
- `server_ip:inet` (required always)

#### `dns_over_tls`
At least one or the other field is required:
- `server_ips:Array(inet)`
- `server_hostname:fqdn`

#### `dns_over_https`
At least one or the other is required:
- `server_ips:Array(inet)`
- `server_uri:URI` (https only)

### TLS considerations
When a user enters a hostname without an IP, we would need to resolve the IP from the Portal in order to pass to connlib. If they enter an IP with a hostname, that's potentially insecure and exposes them to MITM since the cert chain can't be validated, maybe show a warning.

### Port considerations
In nearly all cases the default of `53` for `ip_address`, `853` for `dns_over_tls`, and `443` for `dns_over_https` can be assumed. If we later need to add a port specifier due to customer demand, it should be a different field to make parsing easier. **Note** that connlib expects the IPs to have ports, so the portal will need to append the port number to the IP addresses before sending down.

## tl;dr
For 1.0 I think we can disable the two TLS options and just allow configuring IP addresses -- that should simplify things substantially. cc @conectado

**Comment by AndrewDryga** [comment]
For now, let's do the following steps:
- [x] Change the DNS settings form to also have a type field for each input field
- [x] Change the API response message to include a list of maps instead of a list of strings
- [x] Lock out all unsupported for now options in the dropdown
- [x] Append a default port (53) to the IP-based DNS servers

**Comment by jamilbk** [comment]
Connlib needs updating to support this format. Ended up settling on the following

```json
[{"type": "ip_port", "address": "1.1.1.1:53"} ...]
```
